lib/condor.py

```diff
@@ -188,3 +189,81 @@ def submit_dag(f, vargs, schedd_name, cmd_args=[]):
             print("Execution failed: ", e)
 
     return submit(subfile, vargs, schedd_name)
+
+
+class Job:
+    """
+    Job represents a single HTCondor batch job or cluster with an id like
+        <cluster>[.<process>]@<schedd>
+
+    where if <process> is specified it is a single job, or if not then a cluster
+    of jobs. Examples:
+
+
+        123@schedd.example.com
+        123.0@schedd.example.com
+        123.456@schedd.example.com
+
+    """
+
+    id: str
+    seq: str
+    proc: str
+    schedd: str
+    cluster: bool
```

Comment:
I think this is the first place in the codebase to add type annotations. Thoughts?

Reply:
Why are seq and proc str and not int?

Reply:
I think type annotations are  good; I certainly wouldn't take them out.   Maybe we should make an issue to go back through and type-annotate all our variables?

Reply:
Changed seq and proc to ints.

Reply:
Looks good.